pkg/controller/infrastructure/actuator_delete.go

```diff
@@ -59,12 +59,18 @@ func (a *actuator) Delete(ctx context.Context, log logr.Logger, infra *extension
 	azureClientFactory := NewAzureClientFactory(a.Client())
 	resourceGroupExists, err := infrastructure.IsShootResourceGroupAvailable(ctx, azureClientFactory, infra, config)
 	if err != nil {
-		return err
+		if azureclient.IsAzureAPIUnauthorized(err) {
+			log.Info("failed to check resource group availability due to invalid credentials", err)
```

Comment:
This should be a key value logger, see https://github.com/gardener/gardener-extension-provider-azure/blob/1f43bb5a28d2b67b22517a2cf60853675a20d1cf/vendor/github.com/go-logr/logr/logr.go#L256. 

When there is key without value it should panic with `odd number of arguments passed as key-value pairs for logging`. Can you elaborate how it is working in the current revision?